### PR TITLE
fix: adding .metrics property and circuit breaker metrics

### PR DESCRIPTION
**Pull request by leftieFriele**
First attempt at adding some metrics on the circuit breaking behaviour.
Currently it adds a counter for all events coming from Opossum, we might want to expand on this later.

### CODE REVIEW

File: tests/http-client.test.js [hunk 1]
```diff
@@ -250,7 +250,71 @@ await test('http-client - circuit breaker behaviour', async (t) => {
...
+});
+
+await test('http-client: metrics', async (t) => {
+    await t.test('has a .metrics property', () => {
+        const client = new HttpClient();
+        assert.notEqual(client.metrics, undefined);
+    });
+    await t.test('metric on open breakers', async () => {
+        beforeEach();
+        const url = 'http://lochost:3003';
```
Comment (digitalsadhu) [inline tests/http-client.test.js:262]:
```suggestion
        const url = 'http://localhost:3003';
```

**Review by digitalsadhu — APPROVED** [review]
Looking good

File: lib/http-client.js [hunk 2]
```diff
@@ -66,6 +69,33 @@ export default class HttpClient {
...
+                        name: event,
+                        ...(obj && obj.path && { path: obj.path }),
+                        ...(obj && obj.origin && { origin: obj.origin }),
+                    },
+                });
+            });
+        }
+        // @ts-ignore
+        this.#metrics.on('error', (error) => {
+            this.#logger.error(
```
Comment (wkillerud) [inline lib/http-client.js:94]: Do we want to log at the error level for this? I'm guessing if this errors we have other logs yelling at us 😅

File: lib/http-client.js [hunk 3]
```diff
@@ -66,6 +69,33 @@ export default class HttpClient {
...
+                'Error emitted by metric stream in @podium/client module',
+                error,
```
Comment (wkillerud) [inline lib/http-client.js:96]: Wasn't there something about `pino` and passing in `error` like this as a second argument? Trying to remember.

File: tests/http-client.test.js [hunk 4]
```diff
@@ -250,7 +250,71 @@ await test('http-client - circuit breaker behaviour', async (t) => {
...
+        const client = new HttpClient({
+            threshold: 1,
+            reset: 10,
+            throwOn400: false,
+            throwOn500: false,
+        });
+        const metrics = [];
+        //@ts-expect-error
```
Comment (wkillerud) [inline tests/http-client.test.js:270]: Ideally we shouldn't have type errors here, since it's the public API.

File: lib/http-client.js (hunk 3 above)
Comment (leftieFriele) [reply lib/http-client.js:96]: I think actually it's the opposite, that [abslog](https://github.com/trygve-lie/abslog) says the error should be the first argument actually.

**Pull request merged**